content/blog/testthat-3-2-0/index.Rmd

```diff
@@ -0,0 +1,231 @@
+tags: [testthat, devtools]
+---
+
+```{=html}
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+```
+We're chuffed to announce the release of [testthat](http://testthat.r-lib.org/) 3.2.0.
+testthat makes it easy to turn your existing informal tests into formal, automated tests that you can rerun quickly and easily.
+testthat is the most popular unit-testing package for R, and is used by almost 9,000 CRAN and Bioconductor packages.
+You can learn more about unit testing at <https://r-pkgs.org/tests.html>.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("testthat")
+```
+
+testthat 3.2.0 includes relatively few new features but there have been nine patch releases since testthat 3.1.0.
+These patch releases contained a bunch of experiments that we now believe are ready for the world.
+So this blog post summarises the changes in [3.1.1](https://github.com/r-lib/testthat/releases/tag/v3.1.1), [3.1.2](https://github.com/r-lib/testthat/releases/tag/v3.1.2), [3.1.3](https://github.com/r-lib/testthat/releases/tag/v3.1.3), [3.1.4](https://github.com/r-lib/testthat/releases/tag/v3.1.4), [3.1.5](https://github.com/r-lib/testthat/releases/tag/v3.1.5), [3.1.6](https://github.com/r-lib/testthat/releases/tag/v3.1.6), [3.1.7](https://github.com/r-lib/testthat/releases/tag/v3.1.7), [3.1.8](https://github.com/r-lib/testthat/releases/tag/v3.1.8), [3.1.9](https://github.com/r-lib/testthat/releases/tag/v3.1.9), and [3.1.10](https://github.com/r-lib/testthat/releases/tag/v3.1.10) over the last two years.
+
+Here we'll focus on the biggest news: new expectations, tweaks to the way that error snapshots are reported, support for mocking, a new way to detect if a test has changed global state, and a bunch of smaller UI improvements.
+
+```{r setup}
+#| message: false
+library(testthat)
+```
+
+## Documentation
+
+The first and most important thing to point out is that the second edition of [R Packages](https://r-pkgs.org) contains updated and much expanded coverage of testing.
+Coverage of testing is now split up over three chapters:
+
+-   [Testing basics](https://r-pkgs.org/testing-basics.html)
+-   [Designing your test suite](https://r-pkgs.org/testing-design.html)
+-   [Advanced testing techniques](https://r-pkgs.org/testing-advanced.html)
+
+There's also a new vignette about special files (`vignette("special-files")`) which describes the various special files that you find in `tests/testthat` and when you might need to use them.
```

Comment:
The vignette isn't auto-linking. Does it need the package to be specified?

Reply:
Probably not linking because isn't on CRAN yet? Hopefully will momentarily.

Reply:
Yes!